src/utils.ts

```diff
@@ -378,6 +379,20 @@ class Utils {
     isFirefox(): boolean {
         return typeof(browser) !== "undefined";
     }
+
+    getHash(value: string, times=5000): string {
+        if (times <= 0) return "";
+
+        for (let i = 0; i < times; i++) {
+            let hash = sha256.create();
+            hash.update(value);
+            hash.hex();
+            value = hash.toString();
+        }
+
+        return value;
+    }
```

Comment:
This is great to add, as we'll be able to pre-calculate the public userID on the frontend now.